AstSemantics.md

```diff
@@ -241,14 +234,48 @@ Accesses to global variables specify the index as an integer literal.
 The specification will add atomicity annotations in the future. Currently
 all global accesses can be considered "non-atomic".
 
+## Control flow structures
+
+WebAssembly offers basic structured control flow. All control flow structures
+are statements.
+
+  * `block`: a fixed-length sequence of statements
+  * `if`: if statement
+  * `do_while`: do while statement, basically a loop with a conditional branch
+    (back to the top of the loop)
+  * `forever`: infinite loop statement (like `while (1)`), basically an
+    unconditional branch (back to the top of the loop)
+  * `continue`: continue to start of nested loop
+  * `break`: break to end from nested loop or block
+  * `return`: return zero or more values from this function
+  * `switch`: switch statement with fallthrough
+
+Break and continue statements can only target blocks or loops in which they are
+nested. This guarantees that all resulting control flow graphs are well-structured.
+
+  * Simple and size-efficient binary encoding and compilation.
+  * Any control flow—even irreducible—can be transformed into structured control
+    flow with the
+    [Relooper](https://github.com/kripken/emscripten/raw/master/docs/paper.pdf)
+    [algorithm](http://dl.acm.org/citation.cfm?id=2048224&CFID=670868333&CFTOKEN=46181900),
+    with guaranteed low code size overhead, and typically minimal throughput
+    overhead (except for pathological cases of irreducible control
+    flow). Alternative approaches can generate reducible control flow via node
+    splitting, which can reduce throughput overhead, at the cost of increasing
+    code size (potentially very significantly in pathological cases).
+  * The
+    [signature-restricted proper tail-call](PostMVP.md#signature-restricted-proper-tail-calls)
+    feature would allow efficient compilation of arbitrary irreducible control
+    flow.
```

Comment:
This is more pre-existing text :-). I agree that there's a lot of clarification needed. Let's handle this in follow-up PRs.